packages/astro/src/core/build/static-build.ts

```diff
@@ -470,30 +475,36 @@ export async function copyFiles(fromFolder: URL, toFolder: URL, includeDotfiles
 	);
 }
 
-async function ssrMoveAssets(opts: StaticBuildOptions) {
+async function ssrMoveAssets(opts: StaticBuildOptions, ssrOutputAssetNames: string[]) {
 	opts.logger.info('build', 'Rearranging server assets...');
 	const serverRoot =
 		opts.settings.config.output === 'static'
 			? opts.settings.config.build.client
 			: opts.settings.config.build.server;
 	const clientRoot = opts.settings.config.build.client;
 	const assets = opts.settings.config.build.assets;
-	const serverAssets = new URL(`./${assets}/`, appendForwardSlash(serverRoot.toString()));
-	const clientAssets = new URL(`./${assets}/`, appendForwardSlash(clientRoot.toString()));
+	let serverAssets = new URL(`./${assets}/`, appendForwardSlash(serverRoot.toString()));
+	let clientAssets = new URL(`./${assets}/`, appendForwardSlash(clientRoot.toString()));
+	if (opts?.settings?.config?.vite?.build?.rollupOptions?.output?.assetFileNames) {
```

Comment:
I'm not sure this is the best way to check if the user has defined custom assetFileNames. I am open to suggestions for a better approach.